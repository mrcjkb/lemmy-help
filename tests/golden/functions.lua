local U = { foo = {} }

---NOTE: Local functions are not part of the documentation
---Multiply two integer and print it
---@param this number First number
---@param that number Second number
local function mul(this, that)
    print(this * that)
end

---Add two integer and print it
---
---NOTE: This will be part of the public API
---@param this number First number
---@param that? number
function U.sum(this, that)
    print(this + that or 0)
end

---Subtract second from the first integer
---@param this number
---@param that number Second number
---@return number
---Some secret number that
---we don't know about
---@usage `require('module.U').sub(10, 5)`
function U.sub(this, that)
    return this - that
end

---This is a magical function
---@param this number Non-magical number #1
---@param that number Non-magical number #2
---@return number _ The magical number #1
---@return number #The magical number #2
---and the fun part is the description can span
---
---over mulitple lines and preserves empty lines
---@see U.mul
---@see U.sum
---@see U.sub
U.magical = function(this, that)
    return (U.mul(this, that) / U.sum(that, this)), (U.sum(this, that) * U.sub(that, this))
end

---Function with variable arguments
---@param ... string[]
function U.with_var_arg(...) end

---@param x integer
---@param ... unknown
function U.with_var_arg_end(x, ...) end

---Class method deep access
---@return table
function U.foo:bar()
    self.__index = self
    return setmetatable({}, self)
end

---Method deep access
---@return table
function U.foo.baz()
    return U.foo:bar()
end

---Method with long name and signature just below tag spillover length
function U.very_long_function_name____________() end
---Method with long name and signature just above tag spillover length
function U.very_long_function_name_____________() end
---Method with long param and signature just below tag spillover length
---@param long_param_name______________ string
function U.short_function_name1(long_param_name______________) end
---Method with long param and signature just above tag spillover length
---@param long_param_name_______________ string
function U.short_function_name2(long_param_name_______________) end

return U
