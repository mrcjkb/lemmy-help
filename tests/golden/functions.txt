U.sum({this}, {that?})                                                   *U.sum*
    Add two integer and print it

    NOTE: This will be part of the public API

    Parameters: ~
        {this}   (number)  First number
        {that?}  (number)


U.sub({this}, {that})                                                    *U.sub*
    Subtract second from the first integer

    Parameters: ~
        {this}  (number)
        {that}  (number)  Second number

    Returns: ~
        (number)  Some secret number that
                  we don't know about

    Usage: ~
>lua
        require('module.U').sub(10, 5)
<


U.magical({this}, {that})                                            *U.magical*
    This is a magical function

    Parameters: ~
        {this}  (number)  Non-magical number #1
        {that}  (number)  Non-magical number #2

    Returns: ~
        (number)  The magical number #1
        (number)  The magical number #2
                  and the fun part is the description can span

                  over mulitple lines and preserves empty lines

    See: ~
        |U.mul|
        |U.sum|
        |U.sub|


U.with_var_arg({...})                                           *U.with_var_arg*
    Function with variable arguments

    Parameters: ~
        {...}  (string[])


U.with_var_arg_end({x}, {...})                              *U.with_var_arg_end*

    Parameters: ~
        {x}    (integer)
        {...}  (unknown)


U.foo:bar()                                                          *U.foo:bar*
    Class method deep access

    Returns: ~
        (table)


U.foo.baz()                                                          *U.foo.baz*
    Method deep access

    Returns: ~
        (table)


U.very_long_function_name____________()  *U.very_long_function_name____________*
    Method with long name and signature just below tag spillover length


                                        *U.very_long_function_name_____________*
U.very_long_function_name_____________()
    Method with long name and signature just above tag spillover length


U.short_function_name1({long_param_name______________}) *U.short_function_name1*
    Method with long param and signature just below tag spillover length

    Parameters: ~
        {long_param_name______________}  (string)


                                                        *U.short_function_name2*
U.short_function_name2({long_param_name_______________})
    Method with long param and signature just above tag spillover length

    Parameters: ~
        {long_param_name_______________}  (string)



